lib/src/otx/data/dataset/keypoint_detection_new.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Module for OTXKeypointDetectionDataset."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable, List, Union
+
+import torch
+from torchvision.transforms.v2.functional import to_dtype, to_image
+
+from otx.data.entity.sample import KeypointSample
+from otx.data.transform_libs.torchvision import Compose
+from otx.types.label import LabelInfo
+
+from .base_new import OTXDataset
+
+Transforms = Union[Compose, Callable, List[Callable], dict[str, Compose | Callable | List[Callable]]]
+
+if TYPE_CHECKING:
+    from datumaro.experimental import Dataset
+
+
+class OTXKeypointDetectionDataset(OTXDataset):
+    """OTXDataset class for keypoint detection task."""
+
+    def __init__(self, dm_subset: Dataset, **kwargs) -> None:
+        sample_type = KeypointSample
+        super().__init__(dm_subset=dm_subset, sample_type=sample_type, **kwargs)
```

Comment:
You can simplify by re-ordering a bit:
```suggestion
        dm_subset = dm_subset.convert_to_schema(sample_type)
        super().__init__(dm_subset=dm_subset, sample_type=sample_type, **kwargs)
```